src/app/register/register.component.ts

```diff
@@ -100,9 +104,35 @@ export class RegisterComponent implements OnInit {
     this._titleService.setSubtitle('Register')
   } // OnInit
 
+  ngAfterViewInit() {
+    this.focusError(this.serviceErrors['shibbolethInst'], this.shibbolethInstError)
+    this.focusError(this.serviceErrors['user'], this.userError)
+  }
+
+  private focusError(error: boolean, el: ElementRef) {
+    if (error) {
+      if (el && el.nativeElement) {
+        this.focusErrorElement(el.nativeElement)
+      }
+    }
+  }
+
+  private focusErrorElement(el: HTMLElement) {
+    setTimeout(() => {
```

Comment:
Is it feasible to use retries instead of timeouts to ensure the page is ready as quickly as possible? Adding an extra second seems like a last resort.

Reply:
@momo-sa The use of the settimeout here is not to ensure the page is ready. There is code in other components that set focus after a timeout delay of 100ms. When testing with VoiceOver, the delay of 1000ms was the sweet spot that made it so the narrator didn't stumble on words by being re-focused too quickly. Hope that makes sense! I can do a demo for you if you like.